tests/rules/no-missing-label-refs.test.js

```diff
@@ -149,5 +158,165 @@ ruleTester.run("no-missing-label-refs", rule, {
 				},
 			],
 		},
+		{
+			code: "foo][bar]\n\n[baz]: http://baz.com",
+			errors: [
+				{
+					messageId: "notFound",
+					data: { label: "bar" },
+					line: 1,
+					column: 6,
+					endLine: 1,
+					endColumn: 9,
+				},
+			],
+		},
+		{
+			code: "foo][bar][baz]\n\n[bar]: http://bar.com",
+			errors: [
+				{
+					messageId: "notFound",
+					data: { label: "baz" },
+					line: 1,
+					column: 11,
+					endLine: 1,
+					endColumn: 14,
+				},
+			],
+		},
+		{
+			code: "[foo]\n[foo][bar]",
+			errors: [
+				{
+					messageId: "notFound",
+					data: { label: "foo" },
+					line: 1,
+					column: 2,
+					endLine: 1,
+					endColumn: 5,
+				},
+				{
+					messageId: "notFound",
+					data: { label: "bar" },
+					line: 2,
+					column: 7,
+					endLine: 2,
+					endColumn: 10,
+				},
+			],
+		},
```

Comment:
This was another bug: the regex that looks for `[foo][bar]` matched first and then the loop just skipped the previous code so only one error was reported.